### PR TITLE
Add rmod support to SymIntNode

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #86053
* #86043
* #86046
* #86045
* #86044
* #86042

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

File: torch/csrc/jit/python/init.cpp [hunk 1]
```diff
@@ -1391,6 +1391,12 @@ void initJITBindings(PyObject* module) {
                 auto snb = toSymIntNode(a, b);
                 return a->mod(snb);
               })
+          .def(
+              "__rmod__",
+              [](c10::SymIntNode a, py::object b) -> c10::SymIntNode {
```
Comment (wconstab) [inline torch/csrc/jit/python/init.cpp:1396]: i thought python switched the operands already.  then you'd still want a->mod(snb) wouldn't you?

Comment (ezyang) [reply torch/csrc/jit/python/init.cpp:1396]: Err, no? x % y is x.__mod__(y) or y.__rmod__(y). And we are not reversing before calling into C++. And swapping is symmetric with the other r method bindings in this file

Comment (wconstab) [reply torch/csrc/jit/python/init.cpp:1396]: oh, you are right.

**Review by wconstab — APPROVED** [review]

**Pull request closed without merging**